feat: added sidebar (#1)

ZhangTerrence opened the pull request:
I added the sidebar but there are some features I wasn't able to add:

- No icons in the buttons, apparently ChakraUI requires you to install a separate node package to use its icons. Also, ChakraUI's icon library is pretty small. https://chakra-ui.com/docs/components/icon/usage
- I left the image srcs empty for now
- I don't know how it should look for mobile so right now its not mobile responsive

Also I put the Sidebar components in a newly created components directory. I wasn't sure where to put it for this project it's there for now.

miguel-merlin commented on src/components/Sidebar.tsx:
```diff
@@ -0,0 +1,46 @@
+        <Flex width="100%" height="100%" backgroundColor="blue.500" direction="column" justifyContent="space-between" color="white">
+          <Box>
+            <HStack padding="16px">
+              <Image src='' alt="Blueprint Logo" boxSize="100px"></Image>
```
@ZhangTerrence I just added the logos in your branch ```feat/sidebar``` under ```src/assets/images```. Can you update the src

miguel-merlin commented on src/components/Sidebar.tsx:
```diff
@@ -0,0 +1,46 @@
+            <Box paddingX="8px">
+              <Button onClick={() => { setHidden(!hidden) }} width="100%" colorScheme="whiteAlpha" padding="8px" justifyContent="start" backgroundColor="transparent" variant="solid">Collapse</Button>
+            </Box>
+            <HStack padding="16px">
```
@ZhangTerrence We don't really need this for now. Since authentication is managed by Authelia for now we won't support changing any information of the logged in user

miguel-merlin commented on src/components/Sidebar.tsx:
```diff
@@ -0,0 +1,46 @@
+import { motion } from 'framer-motion'
+import { useState } from 'react'
+
+function Sidebar (): JSX.Element {
```
@ZhangTerrence In ```main``` I just pushed new changes where I use React-Router-Dom to handle routing. Can you add this to the ```HomePage.tsx``` and change the routing of the Sidebar component (When clicked on Home go to HomePage, when clicked on Blog go to BlogPage etc).

ZhangTerrence commented:
I updated the image srcs, added routing functionality to the buttons, and moved Sidebar to HomePage. As for the icons, I was able to get some free svgs so I created custom icons which I temporarily put in the icons directory.

miguel-merlin commented:
You're crazy. Thanks

miguel-merlin reviewed (approved):
Thanks for your work!
